### PR TITLE
Add the `api` value to `event.category`

### LINKED ISSUE #2138: New value for `event.category`: `api`
**Summary**

Elastic Endpoint is working on expanding its data collection capabilities. [Currently existing values](https://www.elastic.co/guide/en/ecs/master/ecs-allowed-values-event-category.html) for the `event.category` field don't match the nature of the events as we're collecting raw information from the host operating system. So using one of the existing values might lead to confusion.

**Motivation**:

We suggest adding a new value for `event.category`: `api`. The new value would be used to categorize information collected from various OS API or logging, and would offer access to the parameters passed to the API. This allows retrieving raw events as they happened on the host. 

**Detailed Design**:

Provide additional details around the design of the proposed changes.

* Field names
* Example values for the fields
  * `event.category: api`
*  `event.type` will use existing values
* Any example events that map to the proposed use case(s)
  Endpoint would use such type for in-memory credential dumping attempts on Windows through the `OpenProcess`/`OpenThread` API calls; or ETW event collections.

### PR DESCRIPTION
**Pull request by calladoum-elastic**
This Pull Request adds a new value for the `event.category` : `api`. Events in this category correspond to API events propagated directly from the Operating System from either the native API function (for instance, OS callbacks from Windows or Linux), system calls, or also managed source of events (such as ETW for instance).

Resolves #2138

### CODE REVIEW

File: schemas/event.yml [hunk 1]
```diff
@@ -156,6 +156,19 @@
       normalize:
         - array
       allowed_values:
+        - name: api
+          description: >
+            Events in this category correspond to API events propagated directly
+            from the Operating System (Windows, Linux, etc.), from either the native
+            API function or system call, or a managed source of events (such as ETW, syslog).
+          expected_event_types:
```
Comment (ebeahan) [inline schemas/event.yml:164]: Narrowing down this list of expected `event.type` values is tough. I could see adding others, but it's also far easier to add others later than remove.

Maybe `info` for APIs that are purely returning information and not taking action or making a state change?

And perhaps also `creation`/`change`/`deletion` for the creating and removing of a resource by way of an API call?

File: schemas/event.yml [hunk 2]
```diff
@@ -156,6 +156,19 @@
       normalize:
         - array
       allowed_values:
+        - name: api
+          description: >
+            Events in this category correspond to API events propagated directly
```
[the remark this answers is not in the document]
Comment (calladoum-elastic) [reply schemas/event.yml:161]: Understood, I'll rephrase

**Review by ebeahan — APPROVED** [review]
LGTM

**Comment by ebeahan** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.7|[](https://github.com/elastic/ecs/pull/2155)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**